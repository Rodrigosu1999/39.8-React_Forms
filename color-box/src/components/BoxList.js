import React, { useState } from "react";
import {v4 as uuid} from "uuid"
import defaultBoxes from "../helpers/defaultBoxes"
import Box from "./Box"
import "../styles/BoxList.css"
import NewBoxForm from "./NewBoxForm";

const BoxList = () => {
    const INITIAL_STATE = defaultBoxes;
    const [boxes, setBoxes] = useState(INITIAL_STATE);

    const addBox = (newBox) => {
        setBoxes(boxes =>[...boxes, {...newBox, id:uuid()}])
    }

    const deleteBox= (id) =>{
        const newBoxList = boxes.filter((box) => box.id !== id);
        setBoxes(newBoxList);
    }

    return (
        <div className="BoxList">
            <h1 className="Boxlist-title">BoxList</h1>
            <div className="BoxList-cards">{boxes.map(box => 
                <Box deleteBox={() => deleteBox(box.id)} id={box.id} color={box.color} height={box.height} width={box.width}/>)}
            </div>
            <NewBoxForm addBox={addBox} />
        </div>
    );
}

export default BoxList;