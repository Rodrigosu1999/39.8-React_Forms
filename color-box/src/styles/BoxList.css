.BoxLis-cards{
    display: flex;
    justify-content: space-around;
    flex-wrap: wrap;
}

.BoxList{
    width: 75%;
    border: 4px solid black;
    margin: auto;
    margin-top: 20px;
}