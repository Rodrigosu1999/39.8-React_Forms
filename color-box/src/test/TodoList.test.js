import React from 'react';
import { fireEvent, queryAllByText, queryByAttribute, render, screen } from '@testing-library/react';
import BoxList from '../components/BoxList';

it('Renders App', () => {
  render(<BoxList />);
});

it('Passes Snapshot', () => {
  const {asFragment} = render(<BoxList />);
  expect(asFragment()).toMatchSnapshot();
});

it('renders correct text', () => {
  const {getByText} = render(<BoxList />);
  console.log(getByText('BoxList'))
});

it('should add new box', () => {
  const {queryByText, getByLabelText, queryAllByText} = render(<BoxList />);
  const colorInput = getByLabelText('Color');
  const heightInput = getByLabelText('Height');
  const widthInput = getByLabelText('Width');
  const btn = queryByText('Add new box!')
  let deleteBtns = queryAllByText('X')

  expect(deleteBtns.length).toEqual(1);

  fireEvent.change(colorInput, {target: {value: "#00FF00"}});
  fireEvent.change(heightInput, {target: {value: 100}});
  fireEvent.change(widthInput, {target: {value: 100}});
  fireEvent.click(btn);
  deleteBtns = queryAllByText('X')

  expect(deleteBtns.length).toEqual(2);
});

it('should delete item', () => {
  const {queryByText, queryAllByText} = render(<BoxList />);
  const btn = queryByText('X')
  let deleteBtns = queryAllByText('X')
  expect(deleteBtns.length).toEqual(1);

  fireEvent.click(btn);
  deleteBtns = queryAllByText('X')

  expect(deleteBtns.length).toEqual(0);
});

