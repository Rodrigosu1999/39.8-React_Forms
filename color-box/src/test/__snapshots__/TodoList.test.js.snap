// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Passes Snapshot 1`] = `
<DocumentFragment>
  <div
    class="BoxList"
  >
    <h1
      class="Boxlist-title"
    >
      BoxList
    </h1>
    <div
      class="BoxList-cards"
    >
      <div
        class="Box"
        id="1"
        style="background-color: rgb(0, 0, 255); height: 100px; width: 100px;"
      >
        <button>
          X
        </button>
      </div>
    </div>
    <form>
      <label
        for="color"
      >
        Color
      </label>
      <input
        id="color"
        name="color"
        type="color"
        value="#FF0000"
      />
      <label
        for="height"
      >
        Height
      </label>
      <input
        id="height"
        max="200"
        min="20"
        name="height"
        type="range"
        value="100"
      />
      <label
        for="width"
      >
        Width
      </label>
      <input
        id="width"
        max="200"
        min="20"
        name="width"
        type="range"
        value="100"
      />
      <button>
        Add new box!
      </button>
    </form>
  </div>
</DocumentFragment>
`;
